src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.ImmutableTypeParameterArgumentAnalysis.cs

```diff
@@ -0,0 +1,642 @@
+﻿using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace D2L.CodeStyle.Analyzers.Immutability;
+public sealed partial class ImmutabilityAnalyzer {
+
+	private static class ImmutableTypeParameterArgumentAnalysis {
+
+		public static void Register(
+			CompilationStartAnalysisContext context,
+			AnnotationsContext annotationsContext,
+			ImmutabilityContext immutabilityContext
+		) {
+			// Type arguments of method return types
+			// Type arguments of method parameters
+			context.RegisterSymbolAction(
+				ctx => {
+					var symbol = (IMethodSymbol)ctx.Symbol;
+
+					// Ignore auto-implemened property methods

```

Comment:
"auto-implemened" was misleading here.  It's all properties because they don't have generics, correct?

Reply:
Updated comment and handling indexers